File: .claude/skills/claude-agent-sdk/references/examples/typescript/simple-query.ts
// Simple Query Example
// Demonstrates basic agent usage with built-in tools

import { query } from '@anthropic-ai/claude-agent-sdk';

async function main() {
  console.log('Starting simple agent query...\n');

  for await (const message of query({
    prompt: 'Analyze the files in the current directory and summarize what this project does',
    options: {
      allowedTools: ['Bash', 'Glob', 'Read'],
      maxBudgetUsd: 0.5,
      maxTurns: 20
    }
  })) {
    // Log text responses
    if (message.type === 'text') {
      console.log('Claude:', message.text);
    }

    // Log tool usage
    if (message.type === 'tool_use') {
      console.log(`\n[Tool] ${message.name}`);
    }

    // Handle final result
    if (message.type === 'result') {
      console.log('\n--- Result ---');

      if (message.subtype === 'success') {
        console.log('Status: Success');
        console.log('Result:', message.result);
        console.log(`Cost: $${message.total_cost_usd.toFixed(4)}`);
        console.log(`Turns: ${message.turn_count}`);
      } else {
        console.error('Status: Error');
        console.error('Error type:', message.subtype);
      }
    }
  }
}

main().catch(console.error);

File: .claude/skills/claude-agent-sdk/references/examples/python/simple_query.py
# Simple Query Example
# Demonstrates basic agent usage with built-in tools

import asyncio
from claude_agent_sdk import query, ClaudeAgentOptions

async def main():
    print("Starting simple agent query...\n")

    async for message in query(
        prompt="Analyze the files in the current directory and summarize what this project does",
        options=ClaudeAgentOptions(
            allowed_tools=["Bash", "Glob", "Read"],
            max_budget_usd=0.5,
            max_turns=20
        )
    ):
        # Log text responses
        if message.type == "text":
            print(f"Claude: {message.text}")

        # Log tool usage
        if message.type == "tool_use":
            print(f"\n[Tool] {message.name}")

        # Handle final result
        if message.type == "result":
            print("\n--- Result ---")

            if message.subtype == "success":
                print("Status: Success")
                print(f"Result: {message.result}")
                print(f"Cost: ${message.total_cost_usd:.4f}")
                print(f"Turns: {message.turn_count}")
            else:
                print("Status: Error")
                print(f"Error type: {message.subtype}")

if __name__ == "__main__":
    asyncio.run(main())
